### ISSUE TITLE
Issue with numba 0.59.0

### ISSUE THREAD

**Issue by israelmcmc** [cositools/cosipy#133]
David Palmore reported an issue installing cosipy related to numba version 0.59.0. It's not really an issue with cosipy, but I'll add it here to remember me to add it to the troubleshooting section in the documentation. David fixed this by downgrading to numba 0.57.0

**Comment by israelmcmc** [comment]
Here's the dependency tree in my environment. It seems numba is needed by astromodels and sparse.

[pipdeptree.txt](https://github.com/cositools/cosipy/files/14212371/pipdeptree.txt)

**Comment by ckarwin** [comment]
Ah, I see, thanks.

**Comment by avalluvan** [comment]
To add, the API `@numba.generated_jit` [has been deprecated](https://github.com/numba/numba/blob/3edb458c69daeaaffbfc938145c367eda1cf043f/docs/upcoming_changes/9353.expired.rst#L2). "Use of `@numba.extending.overload` and the high-level extension API is recommended as a replacement."

**Comment by StevenBoggs** [comment]
Using the command: 
% conda install numba=0.57.0 
Seemed to get me past this hurdle for now....

**Comment by ckarwin** [comment]
Pierre Jean had the same issue with numba. Downgrading to version 0.57.0 also worked for him. Below I copy his initial error message, for reference:

The cosipy installation seemed to be fine (via conda) on my user on the server of Jürgen (linux) but I obtained the following error with the python command "from cosipy import BinnedData"
=====
>>> from cosipy import BinnedData
ImportError: cannot import name 'generated_jit' from 'numba' (/home/jean/.conda/envs/cosipy/lib/python3.10/site-packages/numba/__init__.py)
=====

**Comment by israelmcmc** [comment]
I looked into this a little bit more. The traceback shows the issue propagates from the [interpolation](https://pypi.org/project/interpolation/) package (which call numba's generated_jit, and is called by astromodels). See this relevant issues:
https://github.com/EconForge/interpolation.py/issues/103
https://github.com/EconForge/interpolation.py/issues/110

I made a new release (v0.0.2a4) adding restricting numba<=0.58.0. It will throw a deprecation warning instead of an error.

I'll leave this issue open since we should remove the restriction on the numba version once this is fixed in the dependencies.
